tools/libSwiftScan/libSwiftScan.cpp

```diff
@@ -698,6 +705,31 @@ swiftscan_diagnostics_set_dispose(swiftscan_diagnostic_set_t* diagnostics){
   delete diagnostics;
 }
 
+//=== Source Location -----------------------------------------------------===//
+
+swiftscan_string_ref_t
+swiftscan_source_location_get_buffer_identifier(swiftscan_source_location_t source_location) {
+  return source_location->buffer_identifier;
+}
+
+int64_t
+swiftscan_source_location_get_line_number(swiftscan_source_location_t source_location) {
+  return source_location->line_number;
+}
+
+int64_t
+swiftscan_source_location_get_column_number(swiftscan_source_location_t source_location) {
+  return source_location->column_number;
+}
+
+
+void swiftscan_source_location_dispose(swiftscan_source_location_t source_location) {
```

Comment:
Do we need this API? The only place can return source location is from a diagnostics and if we call this function, it is going to be double free when dispose diagnostics.

Need good documentation for the ownership for the data structures.

Reply:
We do not. Removed!